app/furniture/marketplace/stripe_events_controller.rb

```diff
@@ -0,0 +1,29 @@
+class Marketplace
+  class StripeEventsController < Controller
+    skip_before_action :verify_authenticity_token
+
+    # @todo add signing check using the stripe webhook signing secret
+    # @see https://stripe.com/docs/webhooks/quickstart
+    def create
+      skip_authorization
+      payload = request.body.read
+      signature = request.env["HTTP_STRIPE_SIGNATURE"]
+
+      event = Stripe::Webhook.construct_event(payload, signature, marketplace.stripe_webhook_endpoint_secret)
```

Comment:
@anaulin - Whenever we are forwarding and replaying events, it hangs up on this area; even if we manually copy/paste the webhook signing secret directly during pry.

We haven't been able to figure out why it's not playing nicely, but if you have any insights it would be appreciated. We've:


1. Deleted and recreated the Webhook programmatically 
2. Manually copied in the signing key
3. Set the "Tolerance" to 999999999999 in an attempt to make it support however much clock-drift there may be between local dev and stripe servers

Thoughts?

Reply:
Turns out... the webhook_secret gets generated on a per `stripe listen` basis, so I was setting the secret to the wrong secret. WOMP WOMP WOMP.